src/statement_pruner.cpp

```diff
@@ -1,37 +1,44 @@
 #include "statement_pruner.h"
-#include "pruner_util.h"
 #include "ir/visitor.h"
-
+#include "pruner_util.h"
 
 namespace P4PRUNER {
 
-IR::Node *Pruner::preorder(IR::Statement *s) {
+const IR::Node *Pruner::preorder(IR::Statement *s) {
     for (uint64_t i = 0; i < to_prune.size(); i++) {
         if (*to_prune.at(i) == *s) {
             return nullptr;
         }
     }
     return s;
 }
-IR::Node *Pruner::preorder(IR::BlockStatement *s) {
+
+const IR::Node *Pruner::preorder(IR::BlockStatement *s) {
     for (auto c : s->components) {
         visit(c);
     }
 
-    return (IR::Node *)s;
+    return s;
+}
+
+const IR::Node *Pruner::preorder(IR::ReturnStatement *s) {
```

Comment:
The code sometimes did not compile because a return statement was missing. We can not prune return statements.